test/feature_discovery_test.dart

```diff
@@ -79,40 +79,100 @@ void main() {
+        // overlay 2 and 3 should be displayed or just one of them
+        // depending on allowShowingDuplicate.
+        expect(find.text(texts[0]), findsNothing);
+        expect(find.text(texts[1]),
+            allowShowingDuplicate ? findsNWidgets(2) : findsOneWidget);
+
+        FeatureDiscovery.completeCurrentStep(context);
+        await tester.pumpAndSettle();
+        // Overlays 2 and 3 should have disappeared, and the last overlay should appear.
+        expect(find.text(texts[1]), findsNothing);
+        expect(find.text(texts[2]), findsOneWidget);
+      });
+    }
+
+    testWidgets('Show other overlay after duplicate has been removed',
         (WidgetTester tester) async {
-      await tester.pumpWidget(const TestWidget(featureIds: featureIds));
-      final Finder finder = find.byType(TestIcon);
-      expect(finder, findsNWidgets(featureIds.length));
-      final BuildContext context = tester.firstState(finder).context;
-      texts.forEach((t) => expect(find.text(t), findsNothing));
-      FeatureDiscovery.discoverFeatures(context, steps);
-      await tester.pumpAndSettle();
-      // First overlay should appear.
-      expect(find.text(texts[0]), findsOneWidget);
-      FeatureDiscovery.completeCurrentStep(context);
+      const String featureId = 'feature';
+      const IconData featureIcon = Icons.content_copy;
+
+      String staticFeatureTitle = 'Static',
+          disposableFeatureTitle = 'Disposable';
+
+      await tester.pumpWidget(WidgetWithDisposableFeature(
+        featureId: featureId,
+        featureIcon: featureIcon,
+        staticFeatureTitle: staticFeatureTitle,
+        disposableFeatureTitle: disposableFeatureTitle,
+      ));
+
+      final Finder stateFinder = find.byType(WidgetWithDisposableFeature);
+      expect(stateFinder, findsOneWidget);
+      final WidgetWithDisposableFeatureState state =
+          tester.firstState(stateFinder);
+
+      // Need some widget to return a context that has the Bloc widget as an ancestor.
+      final Finder overlayFinder = find.byType(DescribedFeatureOverlay);
+      final BuildContext context = tester.firstState(overlayFinder).context;
+
+      // Feature titles should only be visible once feature discovery has been started.
+      expect(find.text(staticFeatureTitle), findsNothing);
+      expect(find.text(disposableFeatureTitle), findsNothing);
+
+      FeatureDiscovery.discoverFeatures(context, <String>[featureId]);
       await tester.pumpAndSettle();
-      // First overlay should have disappeared, and overlays 2 and 3 should be displayed.
-      expect(find.text(texts[0]), findsNothing);
-      expect(find.text(texts[1]), findsNWidgets(2));
-      FeatureDiscovery.completeCurrentStep(context);
+
+      // Only one of the overlays should be displayed as allowShowingDuplicate is false.
+      expect(find.byIcon(featureIcon), findsOneWidget);
+      // That overlay should be the disposable one because that should be notified first.
```

Comment:
I don't understand this

Reply:
> That overlay should be the disposable one because that should be notified first.

You mean that?

So, the current `allowShowingDuplicate` implementation will prioritize whatever widget is notified first by `Bloc`. The disposable widget in the test widget is the first item in a `Column` and thus it subscribes to the `Bloc` stream first and therefore receives the event first. It seems that this is deterministic, unlike the previous `InheritedWidget` implementation.  
I will update the comment accordingly. If it was something else that was unclear, do not hesitate to ask!